app/src/main/java/org/ole/planet/myplanet/ui/viewer/WebViewActivity.kt

```
@@ -153,7 +162,19 @@ class WebViewActivity : AppCompatActivity() {
 
     private fun setupAssetLoader(): WebViewAssetLoader? {
         val resourceId = intent.getStringExtra("RESOURCE_ID") ?: return null
-        val directory = File(getExternalFilesDir(null), "ole/$resourceId")
+        val baseDir = File(getExternalFilesDir(null), "ole")
+        val directory = File(baseDir, resourceId)
+
+        val isSafe = try {
+            val canonicalBase = baseDir.canonicalPath
+            val canonicalDir = directory.canonicalPath
+            canonicalDir.startsWith(canonicalBase + File.separator) || canonicalDir == canonicalBase
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)